hasher-matcher-actioner/hmalib/lambdas/fetcher.py

```diff
@@ -366,50 +399,59 @@ def post_apply(self, updated: t.Dict = {}):
         table = dynamodb.Table(self.data_store_table)
 
         for update in updated.values():
-            row: t.List[str] = update.as_csv_row()
-            # example row format: ('<raw_indicator>', '<indicator-id>', '<descriptor-id>', '<time added>', '<space-separated-tags>')
-            # e.g (10736405276340','096a6f9...064f', '1234567890', '2020-07-31T18:47:45+0000', 'true_positive hma_test')
-            new_tags = row[4].split(" ") if row[4] else []
-
-            metadata = PDQSignalMetadata.get_from_signal_and_ds_id(
-                table,
-                int(row[1]),
-                S3ThreatDataConfig.SOURCE_STR,
-                str(self.privacy_group),
-            )
+            if update.indicator_type == "HASH_PDQ":
+                # To remain backwards compatible, only handle PDQ here. SME:
+                # @schatten, @barrettolson
 
-            if metadata:
-                new_pending_opinion_change = self.get_new_pending_opinion_change(
-                    metadata, new_tags
-                )
-            else:
-                # If this is a new indicator without metadata there is nothing for us to update
-                return
-
-            metadata = PDQSignalMetadata(
-                signal_id=row[1],
-                ds_id=str(self.privacy_group),
-                updated_at=datetime.now(),
-                signal_source=S3ThreatDataConfig.SOURCE_STR,
-                signal_hash=row[0],  # note: not used by update_tags_in_table_if_exists
-                tags=new_tags,
-                pending_opinion_change=new_pending_opinion_change,
-            )
-            # TODO: Combine 2 update functions into single function
-            if metadata.update_tags_in_table_if_exists(table):
-                logger.info(
-                    "Updated Signal Tags in DB for indicator id: %s source: %s for privacy group: %d",
-                    row[1],
+                # TODO: Once Signal storage is generified, remove this
+                # conditional and handle multiple indicator_types.
```

Comment:
Not quite understanding the roll out plan to deprecate (HASH_PDQ). Happy to follow up offline

Reply:
Not deprecating HASH_PDQ. You'll notice that we have a record type called `PDQSignalMetadata`.  That obviously can't contain MD5 signals.

2 solutions:
1. Either we make a generic record `SignalMetadata`
2. Create another MD5SignalMetadata

We have to select a solution (outside the scope of this set of PRs). And then update this here. This is a reminder to generify signal storage and handle MD5 and PDQ updates at some point.